src/gc/gcee.cpp

```diff
@@ -297,40 +291,64 @@ void GCHeap::UpdatePostGCCounters()
         GetPerfCounters().m_GC.cTotalReservedBytes = reserved_mem;
     }
 
-    _ASSERTE(FitsIn<size_t>(HeapInfo.HeapStats.FinalizationPromotedSize));
-    _ASSERTE(FitsIn<size_t>(HeapInfo.HeapStats.FinalizationPromotedCount));
-    GetPerfCounters().m_GC.cbPromotedFinalizationMem = static_cast<size_t>(HeapInfo.HeapStats.FinalizationPromotedSize);
-    GetPerfCounters().m_GC.cSurviveFinalize = static_cast<size_t>(HeapInfo.HeapStats.FinalizationPromotedCount);
+    GetPerfCounters().m_GC.timeInGC = (uint32_t)g_TotalTimeInGC;
+    GetPerfCounters().m_GC.timeInGCBase = (uint32_t)_timeInGCBase;
+
+    if (!GetPerfCounters().m_GC.cProcessID)
+        GetPerfCounters().m_GC.cProcessID = (size_t)GetCurrentProcessId();
     
+
+    GetPerfCounters().m_GC.cPinnedObj = total_num_pinned_objects;
+    GetPerfCounters().m_GC.cHandles = total_num_gc_handles;
+    GetPerfCounters().m_GC.cSinkBlocks = total_num_sync_blocks;
+
+#endif // ENABLE_PERF_COUNTERS
+
     // Compute Time in GC
-    PERF_COUNTER_TIMER_PRECISION _currentPerfCounterTimer = GET_CYCLE_COUNT();
+    uint64_t _currentPerfCounterTimer = GCToOSInterface::QueryPerformanceCounter();
 
     g_TotalTimeInGC = _currentPerfCounterTimer - g_TotalTimeInGC;
-    PERF_COUNTER_TIMER_PRECISION _timeInGCBase = (_currentPerfCounterTimer - g_TotalTimeSinceLastGCEnd);
+    uint64_t _timeInGCBase = (_currentPerfCounterTimer - g_TotalTimeSinceLastGCEnd);
 
     if (_timeInGCBase < g_TotalTimeInGC)
         g_TotalTimeInGC = 0;        // isn't likely except on some SMP machines-- perhaps make sure that
                                     //  _timeInGCBase >= g_TotalTimeInGC by setting affinity in GET_CYCLE_COUNT
-                                    
-    while (_timeInGCBase > UINT_MAX) 
+
+    while (_timeInGCBase > 0xffffffff) 
     {
         _timeInGCBase = _timeInGCBase >> 8;
         g_TotalTimeInGC = g_TotalTimeInGC >> 8;
     }
 
     // Update Total Time    
-    GetPerfCounters().m_GC.timeInGC = (uint32_t)g_TotalTimeInGC;
-    GetPerfCounters().m_GC.timeInGCBase = (uint32_t)_timeInGCBase;
+    g_GCCounterData.timeInGC = g_TotalTimeInGC;
+    g_GCCounterData.timeInGCBase = _timeInGCBase;
 
-    if (!GetPerfCounters().m_GC.cProcessID)
-        GetPerfCounters().m_GC.cProcessID = (size_t)GetCurrentProcessId();
-    
     g_TotalTimeSinceLastGCEnd = _currentPerfCounterTimer;
+}
 
-    GetPerfCounters().m_GC.cPinnedObj = total_num_pinned_objects;
-    GetPerfCounters().m_GC.cHandles = total_num_gc_handles;
-    GetPerfCounters().m_GC.cSinkBlocks = total_num_sync_blocks;
-#endif //ENABLE_PERF_COUNTERS
+int GCHeap::GetTimeInGC()
+{
+    if (g_GCCounterData.timeInGCBase == 0)
+    {
+        return 0;
+    }
+    return (int)(g_GCCounterData.timeInGC * 100 / g_GCCounterData.timeInGCBase);
+}
+
+uint64_t GCHeap::GetGenerationSize(int gen)
+{
+    return g_GCCounterData.generationSizes[gen];
+}
+
+uint64_t GCHeap::GetLOHSize()
```

Comment:
>GetLOHSize [](start = 17, length = 10)

why is there a different API for LOH instead of just getting gen3 data?

Reply:
Removed in the latest commit